tests/test_views_admin.py

```diff
@@ -0,0 +1,242 @@
+        self.app.put_json(self.doomed_bucket_url, MINIMALIST_BUCKET, headers=self.doomed_user)
+        self.app.put_json(
+            self.doomed_collection_url, MINIMALIST_COLLECTION, headers=self.doomed_user
+        )
+        # Temporarily grant write to self.user_principal to allow them
+        # to create a collection
+        self.app.patch_json(
+            self.doomed_bucket_url,
+            {"permissions": {"write": [self.user_principal]}},
+            headers=self.doomed_user,
+        )
+        self.app.put_json(
+            self.doomed_bucket_collection_url, MINIMALIST_COLLECTION, headers=self.headers
+        )
+        self.app.patch_json(
+            self.doomed_bucket_url, {"permissions": {"write": []}}, headers=self.doomed_user
+        )
+        self.app.put_json(
+            self.safe_bucket_url,
+            {**MINIMALIST_BUCKET, "permissions": {"write": [self.doomed_user_principal]}},
+            headers=self.headers,
+        )
+        self.app.put_json(
+            self.safe_bucket_doomed_collection_url, MINIMALIST_COLLECTION, headers=self.doomed_user
+        )
+        self.app.put_json(self.safe_collection_url, MINIMALIST_COLLECTION, headers=self.headers)
+        self.app.patch_json(
+            self.safe_collection_url,
+            {"permissions": {"write": [self.doomed_user_principal]}},
+            headers=self.headers,
+        )
+        self.app.put_json(
+            self.safe_collection_doomed_record_url, MINIMALIST_RECORD, headers=self.doomed_user
+        )
+        self.app.patch_json(
+            self.safe_collection_url, {"permissions": {"write": []}}, headers=self.headers
+        )
+        self.app.put_json(
+            self.safe_record_url,
+            {**MINIMALIST_RECORD, "permissions": {"read": [self.doomed_user_principal]}},
+            headers=self.headers,
+        )
+        self.app.put_json(
+            self.shared_bucket_url,
+            {
+                **MINIMALIST_BUCKET,
+                "permissions": {"write": [self.doomed_user_principal, self.user_principal]},
+            },
+            headers=self.doomed_user,
+        )
+        self.app.put_json(
+            self.safe_group_containing_doomed_user,
+            {"data": {"members": [self.doomed_user_principal, self.user_principal]}},
+            headers=self.headers,
+        )
+
+        # Actually do the delete. Each `test_` method checks some
+        # consequence of what got deleted.
+        self.app.delete(self.delete_user_url, headers=self.headers)
+
+    def test_doomed_bucket_was_deleted(self):
+        with self.assertRaises(exceptions.ObjectNotFoundError):
+            self.storage.get("bucket", "", "user_to_delete")
+
+    def test_doomed_bucket_collection_was_deleted(self):
+        # Everything under this bucket should be gone
+        with self.assertRaises(exceptions.ObjectNotFoundError):
+            self.storage.get("collection", self.doomed_bucket_url, "collection")
```

Comment:
I realized (when refactoring the core notions stuff), that using kwargs explicitly helps a lot. (`resource_name=...`)